### PR TITLE
The first revision

### PR DESCRIPTION
**Pull request by dmdashenkov**
This PR adds the code of the example for the very first revision of the **Integrating with a third party** article.

### CODE REVIEW

**Comment by dmdashenkov** [comment]
@armiol, PTAL.

File: .gitmodules [hunk 1]
```diff
@@ -0,0 +1,3 @@
+[submodule "config"]
+	path = config
+	url = https://github.com/SpineEventEngine/config
```
Comment (armiol) [inline .gitmodules]: I would copy this over instead of sub-moduling it.

File: README.md [hunk 2]
```diff
@@ -9,20 +9,15 @@ The repository accompanies the article on third-party integration in Spine poste
 The system consists of several modules. All of the modules represent a way of integrating with
 the **Takeoffs & Landings** Bounded Context.
 
-### Air Traffic Context
-
-The **Air Traffic** Bounded Context is the closest of the all to **Takeoffs & Landings**. The two contexts share a common
-core in the form of the `Aircraft` Aggregate.
-
 ### Airplane Supplies Context
```
Comment (armiol) [inline README.md:17]: I would also put a few words on what each context is.

File: airplane-supplies/src/main/java/io/spine/example/airport/supplies/AirplaneId.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+package io.spine.example.airport.supplies;
+
+import com.google.common.base.MoreObjects;
+import com.google.common.base.Objects;
+
+import java.util.UUID;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+public final class AirplaneId {
```
Comment (armiol) [inline airplane-supplies/src/main/java/io/spine/example/airport/supplies/AirplaneId.java:33]: As always, let's document the public API.

File: airplane-supplies/src/main/java/io/spine/example/airport/supplies/Main.java [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+package io.spine.example.airport.supplies;
+
+import io.grpc.Server;
+import io.grpc.ServerBuilder;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+
+final class Main {
```
Comment (armiol) [inline airplane-supplies/src/main/java/io/spine/example/airport/supplies/Main.java]: Consider calling that in a more meaningful way.

File: airplane-supplies/src/main/java/io/spine/example/airport/supplies/PlaneSupplies.java [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+            this.whenFueled = checkNotNull(whenFueled);
+            return this;
+        }
+
+        public Builder setWhenFrostChecked(Instant whenFrostChecked) {
+            this.whenFrostChecked = checkNotNull(whenFrostChecked);
+            return this;
+        }
+
+        public Builder setWhenPreFlightCheckComplete(Instant whenPreFlightCheckComplete) {
```
Comment (armiol) [inline airplane-supplies/src/main/java/io/spine/example/airport/supplies/PlaneSupplies.java:116]: I would combine the setters for "when complete" and "how successful it was".

File: airplane-supplies/src/main/java/io/spine/example/airport/supplies/SuppliesEventProducer.java [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+import static java.util.Collections.synchronizedList;
+
+/**
+ * Produces events to be consumed by the {@code Takeoffs and Landings} Context.
+ */
+public final class SuppliesEventProducer extends SuppliesEventProducerImplBase {
+
+    private static final FluentLogger log = FluentLogger.forEnclosingClass();
+    @SuppressWarnings("UnsecureRandomNumberGeneration")
+    private static final Random rand = new Random();
```
Comment (armiol) [inline airplane-supplies/src/main/java/io/spine/example/airport/supplies/SuppliesEventProducer.java]: Not sure why not use a secured version.

File: airplane-supplies/src/main/java/io/spine/example/airport/supplies/SuppliesEvents.java [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+import static com.google.common.base.Preconditions.checkNotNull;
+
+final class SuppliesEvents {
+
+    private static final TypeUrl PLANE_FUELED_TYPE =
+            TypeUrl.of(PlaneFueled.class);
+    private static final TypeUrl ANTI_FROSTING_CHECK_COMPLETE_TYPE =
+            TypeUrl.of(AntiFrostingCheckComplete.class);
+    private static final TypeUrl PREFLIGHT_CHECK_COMPLETE_TYPE =
+            TypeUrl.of(AntiFrostingCheckComplete.class);
```
Comment (armiol) [inline airplane-supplies/src/main/java/io/spine/example/airport/supplies/SuppliesEvents.java]: This is not what is is.

File: airplane-supplies/src/main/proto/spine/example/airport/supplies/supplies_events.proto [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+import "spine/options.proto";
+
+option (type_url_prefix) = "type.spine.io";
+option java_package = "io.spine.example.airport.supplies";
+option java_outer_classname = "SuppliesEventsProto";
+option java_multiple_files = true;
+
+message PlaneFueled {
+
+    string plane_id = 1;
```
Comment (armiol) [inline airplane-supplies/src/main/proto/spine/example/airport/supplies/supplies_events.proto]: I would make this a typed thing.

File: takeoffs-and-landings/src/main/java/io/spine/example/airport/tl/PollingClient.java [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.example.airport.tl;
+
+/**
+ * A client which runs indefinitely pulling data from an external API.
```
Comment (armiol) [inline takeoffs-and-landings/src/main/java/io/spine/example/airport/tl/PollingClient.java]: `A client continuously pulling the data from an external API`.

Also, is it `Polling` client or `Pulling`?

**Review by armiol — CHANGES_REQUESTED** [review]

**Comment by dmdashenkov** [comment]
@armiol, PTAL.

**Review by armiol — APPROVED** [review]

**Pull request merged**